NotifyMe/views/views.py

```diff
@@ -0,0 +1,324 @@
+import logging
+from django.core.exceptions import PermissionDenied
+from django.db import IntegrityError
+from django.core.exceptions import ValidationError
+from rest_framework.response import Response
+from ..serializers import UserSerializer, SubscriptionSerializer, SubscriptionPlanSerializer
+from NotifyMe.models.user import User
+from NotifyMe.models.subscription import Subscription
+from NotifyMe.models.subscriptionPlan import SubscriptionPlan
+from rest_framework import status 
+from rest_framework.views import APIView
+from NotifyMe.services.service import UserService, SubscriptionService, SubscriptionPlanService
+
+
+
+logger = logging.getLogger(__name__)
+
+
+# ---------USER-API---------------- 
+# DONOT CALL MODEL INSIDE VIEW---> ONLY CALL SERVICE AND SERIALIZER
+
+def get_response_data(success, message=None, data=None):
+    response_data = {'success': success}
+    if message:
+        response_data['message'] = message
+    if data is not None:
+        response_data['data'] = data
+    return response_data
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+
+     
+class UserAPI(APIView):
+        
+    def get(self, request):
+        user_service = UserService()
+        try:
+            logger.info("Fetching user data")
+            objects = user_service.get_all_users()
+            serializer = UserSerializer(objects, many=True)
+            logger.info("Successfully fetched user data")
+            return Response(get_response_data(True, data=serializer.data))
+        except PermissionDenied:
```

Comment:
why are you catching PermissionDenied since you not throwing it from get__all__user() service method and you are not handling database error as well

Reply:
Database error is handled in .get_all_users() so if any database error will occur they it will be handled there.